src/accessors/IPersistence.ts

```diff
@@ -0,0 +1,28 @@
+import { IPersistenceRead } from './IPersistenceRead';
+/**
+ * An instance of this interface will be constructed by the Rocketlet Manager
+ * which allows the Rocketlet to access its own (generic) persistence.
+ * In order to provide  high-performance access, the associated Rocket.Chat
+ * model provided in `Rocketlet.getRocketChatAssociation()` is considered when
+ * creating the persistence-accessor
+ */
+export interface IPersistence extends IPersistenceRead{
+    /**
+     * Adds a new record representing the state of the Rocketlet
+     * @param referenceId The ID of the instance of the associated model
+     * @param data The Rocketlets' very own data
+     *
+     * @return An ID of the Rocketlet data
+     */
+    create(referenceId: string, data: object): string;
```

Comment:
@graywolf336 
Motivation: At runtime, it will be common that a Rocketlet's data needs to be read based on an `_id` of a model instance (e. g. a `roomId`). This data has to be read with high performance.
I imagine a generic collection at rocket.chat side which is generic and allows to "join" to the origin model (non-unique index with keys `RocketletName, referenceModel, referenceId`) (referenceModel is kind of superfluous for identification since referenceId is a GUID, but might be good for selections in migrations and alike).